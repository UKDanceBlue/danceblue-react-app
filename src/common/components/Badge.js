import { MaterialIcons } from "@expo/vector-icons";
import { ActivityIndicator, Image, StyleSheet, View } from "react-native";
import { Text } from "react-native-elements";

import { useFirebaseStorageUrl } from "../CustomHooks.ts";

/**
 * A badge icon for use with profiles
 */
const Badge = ({
  imageURL, name
}) => {
  const [ imageRef, imageRefError ] = useFirebaseStorageUrl(imageURL);

  return (
    <View style={styles.container}>
      {!imageRef && !imageRefError && <ActivityIndicator size="large" color="blue" />}
      {imageRefError && (
        <MaterialIcons name="image-not-supported" size={styles.icon.width} color="black" />
      )}
      {imageRef && (
        <>
          <Image style={styles.icon} source={{ uri: imageRef }} />
          <Text>{name}</Text>
        </>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    flex: 1,
    justifyContent: "center",
  },
  icon: {
    height: 50,
    width: 50,
  },
});

export default Badge;
