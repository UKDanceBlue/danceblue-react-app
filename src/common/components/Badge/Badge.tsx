import { Image, Text, View } from "native-base";
import { StyleSheet } from "react-native";

/**
 * A badge icon for use with profiles
 */
const Badge = ({
  imageURL, name
}: {
  imageURL: string;
  name?: string;
}) => {
  return (
    <View style={styles.container}>
      <Image style={styles.icon} source={{ uri: imageURL }} alt={name ? `Icon for ${name}` : "Badge Icon"} />
      {name && <Text>{name}</Text>}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    flex: 1,
    justifyContent: "center",
  },
  icon: {
    height: 50,
    width: 50,
  },
});

export default Badge;
