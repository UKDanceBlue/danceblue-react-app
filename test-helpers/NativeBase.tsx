import { render } from "@testing-library/react-native";
import { NativeBaseProvider, NativeBaseProviderProps } from "native-base";
import { ReactElement } from "react";

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
} as const;

/**
 * Editing this function will require an update to test snapshots.
 * @param element The element to test.
 * @param nativeBaseProps
 * @returns The output of *render()*, wrapped with the NativeBaseProvider.
 */
export const renderWithNativeBase = (element: ReactElement, nativeBaseProps: NativeBaseProviderProps = {}) => {
  return render(
    <NativeBaseProvider
      initialWindowMetrics={inset}
      colorModeManager={nativeBaseProps.colorModeManager}
      disableContrastText={nativeBaseProps.disableContrastText}
      theme={nativeBaseProps.theme}
      config={nativeBaseProps.config}
      isSSR={nativeBaseProps.isSSR}
    >
      {element}
    </NativeBaseProvider>
  );
};
