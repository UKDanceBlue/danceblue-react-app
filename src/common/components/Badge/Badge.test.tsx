import { renderWithNativeBase } from "../../../../test-helpers/NativeBase";

import Badge from "./Badge";

describe("<Badge />", () => {
  it("renders correctly", () => {
    const tree = renderWithNativeBase(<Badge imageURL="www.example.com" name="Test" />).toJSON() as unknown;

    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with no name", () => {
    const tree = renderWithNativeBase(<Badge imageURL="www.example.com" />).toJSON() as unknown;

    expect(tree).toMatchSnapshot();
  });
});
