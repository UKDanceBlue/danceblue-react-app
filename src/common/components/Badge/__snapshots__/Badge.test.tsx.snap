// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Array [
        Object {},
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      accessibilityLabel="Icon for Test"
      alt="Icon for Test"
      dataSet={Object {}}
      onError={[Function]}
      source={
        Object {
          "uri": "www.example.com",
        }
      }
      style={
        Array [
          Object {
            "maxWidth": "100%",
          },
          Object {
            "height": 50,
            "width": 50,
          },
        ]
      }
    />
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 21,
          "textDecorationLine": undefined,
        }
      }
    >
      Test
    </Text>
  </View>
</RNCSafeAreaProvider>
`;

exports[`<Badge /> renders correctly with no name 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Array [
        Object {},
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      accessibilityLabel="Badge Icon"
      alt="Badge Icon"
      dataSet={Object {}}
      onError={[Function]}
      source={
        Object {
          "uri": "www.example.com",
        }
      }
      style={
        Array [
          Object {
            "maxWidth": "100%",
          },
          Object {
            "height": 50,
            "width": 50,
          },
        ]
      }
    />
  </View>
</RNCSafeAreaProvider>
`;
